assembly/pom.xml

```diff
@@ -159,6 +169,78 @@
             </target>
           </configuration>
       </plugin>
+      <plugin>
+        <!--
+          Here we download ammonite dependency required for Spark Connect REPL and copy
+          Spark Connect client to target's jars/connect-repl directory. Both jars will
+          only be loaded when we run Spark Connect shell, see also
+          AbstractCommandBuilder.buildClassPath and SPARK-48936.
+        -->
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>get-ammonite-jar</id>
+            <phase>package</phase>
+            <goals>
+              <goal>exec</goal>
+            </goals>
+            <configuration>
+              <executable>${basedir}/../build/mvn</executable>
+              <arguments>
+                <argument>dependency:copy</argument>
```

Comment:
If we change the scope of `ammonite` in `connect-client-jvm` to compile, do we still need to explicitly download these dependencies? I'm not sure if this will bring any negative impacts?

Reply:
`connect-client-jvm` is not the part of the default packaging so `ammonite` wouldn't be available. Another option is shading but it's too much .. ammonite is too large and unstable in my opinion, and we should actually get rid of this when possible.